smart_open/s3.py

```diff
@@ -130,6 +134,7 @@ def parse_uri(uri_as_string):
         ordinary_calling_format=ordinary_calling_format,
         access_id=access_id,
         access_secret=access_secret,
+        ssl_mode=ssl_mode,
```

Comment:
You don't need to pass this as a separate parameter, because it's already included in the URL scheme.

Reply:
How do you mean? Would you prefer the essence of the `ssl_mode` logic to be moved down into `_consolidate_params`?

Reply:
Yes, exactly.